### PR TITLE
[Robot][GE] Robot test for field reorder verification

### PR DESCRIPTION
**Pull request by schlossmj**
# Critical Changes
Adds new test spec for testing reordering/deleting fields, and verifies that the fields are in the correct order on the GE form.

# Changes

# Issues Closed

https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HsTcQAK/view

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+...             robot/Cumulus/resources/GiftEntryPageObject.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             API Check And Enable Gift Entry
+Suite Teardown  Capture Screenshot and Delete Records and Close Browser
+
+*** Keywords ***
+
+Get Template Builder Field Names
+  @{builder_form_fields} =  Get Webelements  xpath://lightning-input[@class="slds-truncate slds-form-element"]//label
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/reorder_template.robot:15]: Good practice to move the element locator in the locators file for later reference or to make the refactoring easier.
Please move the locator to locators file. Please also see if there is a similar element that already exists to make use of it.

File: robot/Cumulus/resources/NPSP.robot [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 Resource       cumulusci/robotframework/Salesforce.robot
 Library        DateTime
+Library        Collections
```
Comment (rjanjanam) [inline robot/Cumulus/resources/NPSP.robot:5]: Is the collections library for the lists? Where are you making use of this ?

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  FOR  ${label}  IN  @{builder_form_fields}
+      ${name} =  Get Text  ${label}
+      Append to List  ${builder_labels}  ${name}
+  END
+  Set Suite Variable  ${builder_labels}
+
+
+Get Template Form Field Names
+  @{gift_form_fields} =  Get Webelements  xpath://label[@class="slds-form-element__label" or @class="slds-form-element__label slds-no-flex"]
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/reorder_template.robot:26]: Same as my above comment. Move to to specific locators file

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  FOR  ${label}  IN  @{gift_form_fields}
+      ${name} =  Get Text  ${label}
+      Append to List  ${form_labels}  ${name}
+  END
+  Set Suite Variable  ${form_labels}
+
+*** Test Cases ***
+
+Reorder and Modify GE Template Fields
+  [Documentation]                       This tests reordering, adding, and deleting sections
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/reorder_template.robot:38]: Is the documentation correct? You are also checking comparing the order of the labels right? If so please enhance you documentation reflecting the test purpose.

**Review by rjanjanam — COMMENTED** [review]
Great effort Matt. Made minor comments. Please take a look

**Review by rjanjanam — APPROVED** [review]

File: robot/Cumulus/resources/NPSP.robot (hunk 2 above)
Comment (schlossmj) [reply robot/Cumulus/resources/NPSP.robot:5]: This is to enable use of the 'Append to List' keyword.

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot (hunk 1 above)
Comment (schlossmj) [reply robot/Cumulus/tests/browser/gift_entry/reorder_template.robot]: Moved this to a keyword.

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot (hunk 3 above)
Comment (schlossmj) [reply robot/Cumulus/tests/browser/gift_entry/reorder_template.robot]: Moved this to a keyword as well.

File: robot/Cumulus/tests/browser/gift_entry/reorder_template.robot (hunk 4 above)
Comment (schlossmj) [reply robot/Cumulus/tests/browser/gift_entry/reorder_template.robot]: Modified this to better reflect the test's purpose.

**Pull request closed without merging**